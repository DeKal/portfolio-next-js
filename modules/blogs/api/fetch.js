import axios from 'axios'
import { API_END_POINT } from '~/modules/core/consts/api'
import { isSuccess, getData, getError } from '~/modules/core/api/selector'

const fetchPostSummary = async () => {
  const result = await axios.get(`${API_END_POINT}/post/all-post/summary`)
  const { data: response } = result
  if (isSuccess(response)) {
    return getData(response)
  }
  const error = getError(response)
  console.error(
    `There was an error with code ${error.code} and message "${error.message}"`
  )
  return []
}

export { fetchPostSummary }
