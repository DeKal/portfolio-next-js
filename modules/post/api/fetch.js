import axios from 'axios'
import { API_END_POINT } from '~/modules/core/consts/api'
import { isSuccess, getData, getError } from '~/modules/core/api/selector'

const fetchPostFromSlug = async slug => {
  const result = await axios.get(`${API_END_POINT}post/${slug}`)
  const { data: response } = result
  if (isSuccess(response)) {
    const data = getData(response)
    return {
      header: data.title || '',
      content: data.content || '',
      authorName: data.author || '',
      date: data.time || '',
      minRead: '6'
    }
  }
  const error = getError(response)
  console.error(
    `There was an error with code ${error.code} and message "${error.message}"`
  )
  return {}
}

export { fetchPostFromSlug }
