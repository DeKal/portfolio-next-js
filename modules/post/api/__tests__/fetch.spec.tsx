import { fetchPostFromSlug } from '../fetch'
import mockAxios from 'axios'
jest.mock('axios')

const post = {
  data: {
    title: 'Search: The Whole Story',
    content:
      'Query understanding and relevance are key aspects of search, but they don’t tell the whole story.',
    author: 'Daniel Tunkelang',
    time: 'Mar 25, 2019',
    minRead: '6'
  },
  success: true
}

describe('Fetch ', () => {
  it('fetchPostFromSlug with mock request', async () => {
    //@ts-ignore
    mockAxios.get.mockResolvedValue({
      data: post
    })

    const data = await fetchPostFromSlug()
    expect(data).toEqual({
      header: 'Search: The Whole Story',
      content: 'Query understanding and relevance are key aspects of search, but they don’t tell the whole story.',
      authorName: 'Daniel Tunkelang',
      date: 'Mar 25, 2019',
      minRead: '6'
    })
  })

  it('fetchPostFromSlug with empty request', async () => {
    //@ts-ignore
    mockAxios.get.mockResolvedValue({
      data: {
        data: {},
        success: true
      }
    })

    const data = await fetchPostFromSlug()
    expect(data).toEqual({
      header: '',
      content: '',
      authorName: '',
      date: '',
      minRead: '6'
    })
  })

  it('fetchPostFromSlug with error request', async () => {
    //@ts-ignore
    mockAxios.get.mockResolvedValue({
      data: {
        error: {
          code: 204,
          message: 'Cannot find post'
        },
        success: false
      }
    })

    const data = await fetchPostFromSlug()
    expect(data).toMatchSnapshot()
  })
})
