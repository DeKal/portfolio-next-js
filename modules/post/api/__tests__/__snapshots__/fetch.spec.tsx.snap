// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch  fetchPostFromSlug with error request 1`] = `Object {}`;
